### ISSUE TITLE
Integration with NextJs - Actions not being dispatched

### ISSUE THREAD

**Issue by yuschick** [ctrlplusb/easy-peasy#626]
This will not be a very detailed issue, but I feel like I'm at a wall currently with where else to turn.

I have started integrating a new Easy-Peasy store (v4.0.1) into a NextJs (v10.0.6) project. I had no issues using EP with Gatsby before, so I didn't expect any issues here either. However, I think there is some weird magic happening with Next's Fast Refresh in a local development environment.

Ultimately, whenever I do a hard refresh on Page 1, the EP store loads correctly and all functionality works as expected. Actions are dispatched, state is updated, and components are rendered.

However, I then navigate to Page 2 and things get weird.

On Page 2, actions, especially hose in effects hooks at the component level, are not shown in the Redux Dev Tools, nor do they update the state, or trigger renders with the latest information. However, I have placed console logs inside of these actions and thunks and can verify that they are, in fact, called and with the correct data.

There looks to be a disconnect somewhere with the dispatching, maybe, of the actions.

I know this is likely more of a NextJs issue, but I wanted to check here as well and see if there are any more recent discussions or integration concerns since the closed 2019 issues.

**Comment by binajmen** [comment]
Hello @yuschick, I posted a new issue (#627) as I didn't see yours and was stuck with 2019 issues since two days. This is literally the same subject.. Therefore, your input would be more than welcome!

I would be especially interested to understand how you manage the store at server side and/or have your feedback on my small example.

**Comment by yuschick** [comment]
Hey @binajmen. I ended up following this [example](https://github.com/BjoernRave/with-easy-peasy).

Ultimately, the state store in my case is installed as an external library, which slightly complicated the implementation. Ultimately, though, it seems like the issue was passing the store between the server and client to ensure it stayed in sync and initialized.

This isn't my strongest area of understanding, though. Mostly just threw some stuff at the wall and hoped for the best. 🤣

**Comment by binajmen** [comment]
Yes, I saw this example too. However, as it is 2 years old and technogolies like next, redux & easy-peasy evolve fast, I wanted to make sure I have a solution up-to-date before using it in production.

I ended up with two reliable sources to build such a solution: the official example from [Next.js](https://github.com/vercel/next.js/tree/canary/examples/with-redux-wrapper) and the [next-redux-wrapper](https://github.com/kirill-konshin/next-redux-wrapper) repo.

Some interesting stuff I learned:
* How to avoid being opted out of Automatic Static Optimization
* The importance of [state reconciliation during hydration](https://github.com/kirill-konshin/next-redux-wrapper#state-reconciliation-during-hydration) (if you want to hydrate the store server-side)

> Ultimately, though, it seems like the issue was passing the store between the server and client to ensure it stayed in sync and initialized.

The last point might cover this issue.
